### ISSUE TITLE
Pin package level coordinate data

### ISSUE THREAD

**Issue by info-rchitect** [Origen-SDK/origen#298]
All,

When re-writing our pin importer, I realized there is a lot of package coordinate metadata that really has no way to be added to the model.  The current package scoped attributes list is:

~~~ruby
[:location, :dib_assignment, :dib_meta]
~~~

Should we add a `:pkg_meta` attribute similar to the existing functionally scoped `:meta` attribute?

**Comment by ginty** [comment]
It does seem like a good idea, however should it not just be `:meta` ?

Currently it looks like you can supply `:meta` when defining the pin, and then again when defining functions and Origen will merge them into a single `my_pin.meta` attribute depending on what function is enabled.

So I would then expect for the package that I could do:

~~~
add_pin :tdi, packages: { bga: { meta: { blah: :blah } } } 
~~~

And the package specific meta data would get merged into `my_pin.meta` in the same way for the current package.

**Comment by info-rchitect** [comment]
@ginty I didn't know it did the merge.  Should this be possible?

~~~ruby
add_pin :tdi, meta: { die_level_attr: die_level_val }, packages: { bga: { meta: { blah: :blah } } } 
~~~

**Comment by info-rchitect** [comment]
@ginty I was not able to get the following to work:

~~~ruby
dut.add_pin :tdi, packages: { bga: { meta: { blah: :blah } } }
~~~

The metadata was always an empty hash even when I set the package.  I was able to assign the meta at the die level scope.

**Comment by ginty** [comment]
@info-rchitect, yes it doesn't work yet, I'm saying that this is the way it should work if we add something here.

It does work that way already for functions, so this is the way we should make it work for packages also - i.e. rather than adding a `:pkg_meta` attribute which was the original proposal.

**Comment by info-rchitect** [comment]
@ginty OK gotcha, I will make a PR once I get the importer done.
